racetrack_commons/racetrack_commons/plugin/engine.py

```diff
@@ -226,37 +230,50 @@ def write_plugin_config(self, name: str, version: str, config: str):
         self._load_plugins()
         
     def delete_plugin_by_version(self, name: str, version: str):
-        plugin_data = self.find_plugin(name, version)
-        self._delete_plugin(plugin_data)
-        logger.info(f'Plugin {plugin_data.plugin_manifest.name} {plugin_data.plugin_manifest.version} ({plugin_data.zip_path}) has been deleted')
+        try:
+            plugin_data = self.find_plugin(name, version)
+            if self._delete_plugin_files(plugin_data.zip_path.stem):
+                logger.info(f'Plugin {plugin_data.plugin_manifest.name} {plugin_data.plugin_manifest.version} ({plugin_data.zip_path}) has been deleted')
+            else:
+                logger.warning(f'Plugin files were not found: {plugin_data.zip_path.stem}')
+        except EntityNotFound as e:
+            if self._delete_plugin_files(f'{name}-{version}'):
+                logger.info(f'Plugin files have been deleted: {name} {version}')
+            else:
+                raise e
         self._record_last_change()
         self._load_plugins()
 
     def _delete_older_plugin_version(self, plugin_name: str, plugin_version: str):
         try:
             plugin_data = self.find_plugin(plugin_name, plugin_version)
-            self._delete_plugin(plugin_data)
-            logger.info(f'Older plugin version has been deleted: {plugin_data.zip_path.name}')
+            if self._delete_plugin_files(plugin_data.zip_path.stem):
+                logger.info(f'Older plugin version has been deleted: {plugin_data.zip_path.name}')
+            else:
+                logger.warning(f'Plugin files were not found: {plugin_data.zip_path.stem}')
         except EntityNotFound:
-            return
+            if self._delete_plugin_files(f'{plugin_name}-{plugin_version}'):
+                logger.info(f'Older plugin files have been deleted: {plugin_name} {plugin_version}')
 
     def _delete_older_plugins(self, plugin_name: str):
         plugins_data = self.find_plugins(plugin_name)
         for plugin_data in plugins_data:
-            self._delete_plugin(plugin_data)
-            logger.info(f'Older plugin version has been deleted: {plugin_data.zip_path.name}')
-
-    def _delete_plugin(self, plugin_data: PluginData):
-        if plugin_data.zip_path.is_file():
-            plugin_data.zip_path.unlink()
-        else:
-            logger.warning(f'ZIP plugin was not found: {plugin_data.zip_path}')
+            if self._delete_plugin_files(plugin_data.zip_path.stem):
+                logger.info(f'Older plugin version has been deleted: {plugin_data.zip_path.name}')
+            else:
+                logger.warning(f'Plugin files were not found: {plugin_data.zip_path.stem}')
 
-        extracted_dir = Path(self.plugins_dir) / EXTRACTED_PLUGINS_DIR / plugin_data.zip_path.stem
+    def _delete_plugin_files(self, stem: str) -> bool:
+        extracted_dir = Path(self.plugins_dir) / EXTRACTED_PLUGINS_DIR / stem
```

Comment:
I noticed that `plugins_dir` is casted to Path with each use, maybe it could be done once during object init.